### PR TITLE
Allow cubed arrays to be passed to flox groupby

### PR DESCRIPTION
**Pull request by TomNicholas**
Generalizes a small check for chunked arrays in groupby so it now allows cubed arrays through to flox rather than just dask arrays. Does not actually mean that flox groupby will work with cubed yet though, see https://github.com/tomwhite/cubed/issues/223 and https://github.com/xarray-contrib/flox/issues/224

- [x] Should have been done in #7019
- [ ] ~~Tests added~~ (The place to test this would be in [`cubed-xarray`]
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`
- [ ] ~~New functions/methods are listed in `api.rst`~~

### CODE REVIEW

File: xarray/util/generate_aggregations.py [hunk 1]
```diff
@@ -22,12 +22,13 @@
 
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, Any, Callable, Sequence
+from collections.abc import Sequence
+from typing import TYPE_CHECKING, Any, Callable
```
Comment (TomNicholas) [inline xarray/util/generate_aggregations.py:26]: This change was supposed to be a separate commit, but this was only because ruff raised an error after regenerating the `_aggregations.py` file.

**Review by dcherian — APPROVED** [review]
